src/Microsoft.Tye.Core/ConfigModel/ConfigFactory.cs

```diff
@@ -88,5 +96,16 @@ private static ConfigApplication FromYaml(FileInfo file)
             using var parser = new YamlParser(file);
             return parser.ParseConfigApplication();
         }
+
+        private static ConfigApplication FromJson(FileInfo file)
+        {
+            var converter = new ExpandoObjectConverter();
+            dynamic obj = JsonConvert.DeserializeObject<ExpandoObject>(file.OpenText().ReadToEnd(), converter);
+            var serializer = new YamlDotNet.Serialization.Serializer();
+            string yamlContent = serializer.Serialize(obj);
+
+            using var parser = new YamlParser(yamlContent);
+            return parser.ParseConfigApplication();
```

Comment:
Is this step even needed? (would YamlParser just parse the JSON document since JSON is YAML).

I'm thinking about things like line-numbers matching up if we do a conversion step. We just did a bunch of work in the yaml parser/validator to get better error messages with line numbers.

Reply:
Good question, would have to test. First though is no. Yaml is format (space) specific where JSON can be one line no spaces and still be valid.